src/fheroes2/dialog/dialog_selectfile.cpp

```diff
@@ -461,6 +608,14 @@ namespace
                     result = System::concatPath( Game::GetSaveDir(), filename + Game::GetSaveFileExtension() );
                 }
             }
+            else if ( le.MouseClickLeft( nameHeaderRoi ) && settings.GetSaveFileSortingMethod() != SaveFileSortingMethod::FILENAME ) {
+                listUpdated = true;
+                needRedraw = switchFileSorting( settings.GetSaveFileSortingMethod() == SaveFileSortingMethod::FILENAME );
```

Comment:
We can pass `false` here because in the line 611 `settings.GetSaveFileSortingMethod()` is already checked.
```suggestion
                needRedraw = switchFileSorting( false );
```